superset/assets/src/visualizations/PlaySlider.jsx

```diff
@@ -87,7 +89,11 @@ export default class PlaySlider extends React.PureComponent {
     if (this.props.disabled) {
       return;
     }
-    let values = this.props.values.map(value => value + this.increment);
+    let values = this.props.values;
+    if (!Array.isArray(values)) {
```

Comment:
The code in `if` block is not used? Because `values` is set again right after the block.

Reply:
Ah, good point, will fix.

Reply:
It can be confusing if `values` is the new one or the old one.
I prefer to avoid using `let` and try not to mutate variable if not necessary.

```js
const { start, end, step, values, disabled } = this.props;
if (disabled) {
  return;
}
const newValues = (Array.isArray(values) ? values : [values, values + step])
  .map(v => v + this.increment);
// What does cr stand for?
const cr = (newValues[1] > end) ? (newValues[0] - start) : 0;
this.props.onChange(cr === 0 ? newValues : newValues.map(v => v - cr));
```

Reply:
CR is carriage return — how much the controls move back once they hit the end of the slider. I'll make the code more clear, thanks!